pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -902,3 +903,36 @@ func ValidateValidatingAdmissionPolicyUpdate(newC, oldC *admissionregistration.V
 func ValidateValidatingAdmissionPolicyBindingUpdate(newC, oldC *admissionregistration.ValidatingAdmissionPolicyBinding) field.ErrorList {
 	return validateValidatingAdmissionPolicyBinding(newC)
 }
+
+func validateValidatingAdmissionPolicyStatus(status *admissionregistration.ValidatingAdmissionPolicyStatus, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	allErrors = append(allErrors, validateTypeChecking(status.TypeChecking, fldPath.Child("typeChecking"))...)
+	allErrors = append(allErrors, metav1validation.ValidateConditions(status.Conditions, fldPath.Child("conditions"))...)
+	return allErrors
+}
+
+func validateTypeChecking(typeChecking *admissionregistration.TypeChecking, fldPath *field.Path) field.ErrorList {
+	if typeChecking == nil {
+		return nil
+	}
+	return validateExpressionWarnings(typeChecking.ExpressionWarnings, fldPath.Child("expressionWarnings"))
+}
+
+func validateExpressionWarnings(expressionWarnings []admissionregistration.ExpressionWarning, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	for i, warning := range expressionWarnings {
+		allErrors = append(allErrors, validateExpressionWarning(&warning, fldPath.Index(i))...)
+	}
+	return allErrors
+}
+
+func validateExpressionWarning(expressionWarning *admissionregistration.ExpressionWarning, fldPath *field.Path) field.ErrorList {
+	var allErrors field.ErrorList
+	if expressionWarning.Warning == "" {
+		allErrors = append(allErrors, field.Required(fldPath.Child("warning"), "must be non-empty"))
+	}
+	if expressionWarning.FieldRef == "" {
+		allErrors = append(allErrors, field.Required(fldPath.Child("fieldRef"), "must be non-empty"))
```

Comment:
Validate that the value is a JSON Path (to only one of the supported fields) here?

Reply:
Do we have a library to parse this PATH?

Reply:
Check if it is a jsonpath makes sense. I'll suggest let's not put validation on if it is one of the supported field here just to keep the door open.

Reply:
For rollback, yeah, safer is it's just validated to be json path

Reply:
I've update this part so that no further enforcement beyond valid JSONPATH.